be1-go/channel/digitalCash/mod.go

```diff
@@ -0,0 +1,225 @@
+package digitalCash
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"github.com/rs/zerolog"
+	"golang.org/x/xerrors"
+	"popstellar/channel"
+	"popstellar/channel/registry"
+	"popstellar/inbox"
+	jsonrpc "popstellar/message"
+	"popstellar/message/answer"
+	"popstellar/message/messagedata"
+	"popstellar/message/query"
+	"popstellar/message/query/method"
+	"popstellar/message/query/method/message"
+	"popstellar/network/socket"
+	"popstellar/validation"
+	"strconv"
+)
+
+const (
+	msgID = "msg id"
+)
+
+// Channel is used to handle election messages.
+type Channel struct {
+	sockets   channel.Sockets
+	inbox     *inbox.Inbox
+	hub       channel.HubFunctionalities
+	channelID string
+	registry  registry.MessageRegistry
+
+	log zerolog.Logger
+}
+
+// NewChannel returns a new initialized digitalCash channel
+func NewChannel(channelID string, hub channel.HubFunctionalities,
+	log zerolog.Logger) channel.Channel {
+
+	box := inbox.NewInbox(channelID)
+
+	log = log.With().Str("channel", "digitalCash").Logger()
+
+	// Saving on Digital Cash channel too so it self-contains the entire cash history
+	retChannel := &Channel{
+		sockets:   channel.NewSockets(),
+		inbox:     box,
+		channelID: channelID,
+		hub:       hub,
+		log:       log,
+	}
+
+	retChannel.registry = retChannel.NewDigitalCashRegistry()
+
+	return retChannel
+}
+
+// ---
+// Publish-subscribe / channel.Channel implementation
+// ---
+
+// Subscribe is used to handle a subscribe message from the client
+func (c *Channel) Subscribe(socket socket.Socket, msg method.Subscribe) error {
+	c.log.Info().Str(msgID, strconv.Itoa(msg.ID)).Msg("received a subscribe")
+	c.sockets.Upsert(socket)
+
+	return nil
+}
+
+// Unsubscribe is used to handle an unsubscribe message.
+func (c *Channel) Unsubscribe(socketID string, msg method.Unsubscribe) error {
+	c.log.Info().Str(msgID, strconv.Itoa(msg.ID)).Msg("received an unsubscribe")
+
+	ok := c.sockets.Delete(socketID)
+	if !ok {
+		return answer.NewError(-2, "client is not subscribed to this channel")
+	}
+
+	return nil
+}
+
+// Publish handles publish messages for the consensus channel
+func (c *Channel) Publish(publish method.Publish, socket socket.Socket) error {
+	c.log.Info().
+		Str(msgID, strconv.Itoa(publish.ID)).
+		Msg("received a publish")
+
+	err := c.verifyMessage(publish.Params.Message)
+	if err != nil {
+		return xerrors.Errorf("failed to verify publish message on an "+
+			"digital cash channel: %w", err)
```

Comment:
The logger already tells that this is the digital cash channel:
```suggestion
		return xerrors.Errorf("failed to verify publish message: %v", err)
```